src/learner.cc

```diff
@@ -1035,16 +1035,18 @@ class LearnerImpl : public LearnerIO {
     os << '[' << iter << ']' << std::setiosflags(std::ios::fixed);
     if (metrics_.size() == 0 && tparam_.disable_default_eval_metric <= 0) {
       auto warn_default_eval_metric = [](const std::string& objective, const std::string& before,
-                                         const std::string& after) {
-        LOG(WARNING) << "Starting in XGBoost 1.3.0, the default evaluation metric used with the "
-                     << "objective '" << objective << "' was changed from '" << before
-                     << "' to '" << after << "'. Explicitly set eval_metric if you'd like to "
-                     << "restore the old behavior.";
+                                         const std::string& after, const std::string& version) {
+        LOG(WARNING) << "Starting in XGBoost " << version << ", the default evaluation metric "
```

Comment:
Note. We should plan to remove this warning soon. Probably right after 1.4.0 ?

Reply:
Does the warning go to `logging` or stdout?

Reply:
@hcho3 stdout

Reply:
That's the `_log_calback` function in `xgboost/core.py`.

Reply:
Got it. We could modify `_log_callback` to use the `logging` module. Alternatively, we could simply remove the warning message.

Reply:
We can do both in the future, but it will be another breaking change.  To use `logging`, one needs to know the verbosity.  So `_log_callback` should have 2 parameters `_log_callback(msg: str, verbosity: int)` to indicate the level of current message.

Reply:
Got it. Let's remove the warning message after 1.4.0 release.